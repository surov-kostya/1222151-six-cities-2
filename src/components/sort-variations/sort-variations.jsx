import React, {PureComponent} from 'react';
import {func, arrayOf} from 'prop-types';
import {variantType} from '../../models/index';

class SortVariations extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      isSortMenuShown: false
    };
    this.sortVariations = props.sortVariations;
    this.onSort = props.onSort;

    this._handleDropdownToggle = this._handleDropdownToggle.bind(this);
  }

  render() {
    return (
      <div className="places__sorting" action="#" method="get">
        <span className="places__sorting-caption">Sort by</span>
        <button className="places__sorting-type" tabIndex="0"
          style={{border: `none`, backgroundColor: `transparent`, outline: `none`}}
          onClick={this._handleDropdownToggle}>
          {this.props.activeSortVariant && this.props.activeSortVariant.name}
          <svg className="places__sorting-arrow" width="7" height="4">
            <use xlinkHref="#icon-arrow-select"></use>
          </svg>
        </button>
        {this.state.isSortMenuShown && <ul className="places__options places__options--custom places__options--opened">
          {this.sortVariations.map((variant, i) => (
            <li
              key={variant.id}
              onClick={() => {
                this.onSort(variant.id);
                this._handleDropdownToggle();
              }}
              className={
                variant.id === this.props.activeSortVariant.id
                  ? `places__option places__option--active`
                  : `places__option`
              }
              tabIndex={i}
            >
              {variant.name}
            </li>
          ))}
        </ul>}
      </div>
    );
  }

  _handleDropdownToggle() {
    this.setState((prevState) => ({isSortMenuShown: !prevState.isSortMenuShown}));
  }
}

SortVariations.propTypes = {
  activeSortVariant: variantType,
  onSort: func,
  sortVariations: arrayOf(variantType)
};

export default SortVariations;
