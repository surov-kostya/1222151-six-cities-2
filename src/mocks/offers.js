import {shape, number, string, arrayOf} from 'prop-types';

export const placeType = shape({
  id: number,
  name: string,
  imageSrc: string,
  contacts: shape({eMail: string}),
  gallerySrcs: arrayOf(string),
  mark: string,
  rating: number,
  parametres: shape({
    entire: string,
    bedrooms: number,
    adults: number
  }),
  price: number,
  features: arrayOf(string),
  host: shape({
    id: number,
    name: string,
    status: string
  }),
  hostsReview: arrayOf(string)
});

export const places = [
  {
    id: 0,
    name: `Beautiful && luxurious apartment at great location`,
    imageSrc: `img/apartment-01.jpg`,
    contacts: {
      eMail: `Oliver.conner@gmail.com`
    },
    gallerySrcs: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/map.jpg`],
    mark: `Premium`,
    rating: 4.8,
    parametres: {
      entire: `Entire place`,
      bedrooms: 3,
      adults: 4
    },
    price: 120,
    features: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    host: {
      id: 100,
      name: `Angelina`,
      status: `Pro`
    },
    hostsReview: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ]
  },
  {
    id: 1,
    name: `Wood and stone place`,
    imageSrc: `img/apartment-02.jpg`,
    contacts: {
      eMail: `Oliver.conner@gmail.com`
    },
    gallerySrcs: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
    mark: `Premium`,
    rating: 4.8,
    parametres: {
      entire: `Entire place`,
      bedrooms: 3,
      adults: 4
    },
    price: 120,
    features: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    host: {
      id: 100,
      name: `Angelina`,
      status: `Pro`
    },
    hostsReview: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ]
  },
  {
    id: 2,
    name: `Nice, cozy, warm big bed apartment`,
    imageSrc: `img/apartment-03.jpg`,
    contacts: {
      eMail: `Oliver.conner@gmail.com`
    },
    gallerySrcs: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
    mark: `Premium`,
    rating: 4.8,
    parametres: {
      entire: `Entire place`,
      bedrooms: 3,
      adults: 4
    },
    price: 120,
    features: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    host: {
      id: 100,
      name: `Angelina`,
      status: `Pro`
    },
    hostsReview: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ]
  },
  {
    id: 3,
    name: `Canal View Prinsengracht`,
    imageSrc: `img/room.jpg`,
    contacts: {
      eMail: `Oliver.conner@gmail.com`
    },
    gallerySrcs: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
    mark: `Premium`,
    rating: 4.8,
    parametres: {
      entire: `Entire place`,
      bedrooms: 3,
      adults: 4
    },
    price: 120,
    features: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    host: {
      id: 100,
      name: `Angelina`,
      status: `Pro`
    },
    hostsReview: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ]
  }
];
