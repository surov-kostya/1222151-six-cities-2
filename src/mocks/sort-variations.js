export const variations = [
  {
    id: 0,
    name: `Popular`
  },
  {
    id: 1,
    name: `Price: low to high`
  },
  {
    id: 2,
    name: `Price: high to low`
  },
  {
    id: 3,
    name: `Top rated first`
  }
];
