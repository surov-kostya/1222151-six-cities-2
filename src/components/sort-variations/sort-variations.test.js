import React from 'react';
import renderer from 'react-test-renderer';
import SortVariations from 'sort-variations';
import {variations} from '../../mocks/sort-variations';

it(`SortVariations rednders correct`, () => {
  const tree = renderer.create(
      <SortVariations sortVariations={variations} activeSortVariant={variations[0]} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
