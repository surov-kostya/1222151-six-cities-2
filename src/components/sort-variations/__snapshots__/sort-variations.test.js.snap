// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortVariations rednders correct 1`] = `
<div
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <button
    className="places__sorting-type"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "border": "none",
        "outline": "none",
      }
    }
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </button>
</div>
`;
